Source Code/Ticket Management System/supportgiver_menu.c

```diff
@@ -77,30 +104,74 @@ void browseTickets()
 					}
 					case('2'): // Filter by tier
 					{
-						// TODO
+						wchar_t* cty;
+						wchar_t tier[32];
+						wprintf(L"%s\n", L"please input tier");
+						wscanf(L"%s", tier);
+						wcstok(tier, "\n", &cty);
+						struct Vector* filtered_vector = vector_filter(tickets, filterByTier, tier);
+						printTicketsArray(filtered_vector);
+						vector_destroy(filtered_vector);
 						break;
 					}
 					case('3'): // Filter by type
 					{
-						// TODO
+						wchar_t* ctz;
+						wchar_t type[32];
+						wprintf(L"%s\n", L"please input type");
+						wscanf(L"%s", type);
+						wcstok(type, "\n", &ctz);
+						struct Vector* filtered_vector = vector_filter(tickets, filterByType, type);
+						printTicketsArray(filtered_vector);
+						vector_destroy(filtered_vector);
 						break;
 					}
-					case('4'): // Filter by stakeholders
+					case('4'): // sort by severity
```

Comment:
Case numbers changed. Was menu creation updated to reflect new meaning of inputs?